task22/js/main.js

```diff
@@ -1,151 +1,195 @@
-            tableBodyFragment.appendChild(tableRowElement).appendChild(tableDataElement);
-        }
+  const tbodyElement = document.createElement("tbody");
+  tbodyElement.id = "js-tableBody";
+  const tableBodyFragment = document.createDocumentFragment();
+  for (const user of userContentsData) {
+    const tableRowElement = document.createElement("tr");
+    for (const column of Object.keys(userTableColumn)) {
+      const tableDataElement = createElementWithClassName("td", "table-data");
+      tableDataElement.textContent = user[column];
+      tableBodyFragment
+        .appendChild(tableRowElement)
+        .appendChild(tableDataElement);
     }
-    tbodyElement.appendChild(tableBodyFragment);
-    return tbodyElement;
+  }
+  tbodyElement.appendChild(tableBodyFragment);
+  return tbodyElement;
 };
 
-const getCreatedSortButton = () => {
-    const sortButton = document.createElement("button");
-    const sortImage = createElementWithClassName("img", "sort-image");
-    sortButton.id = "js-sortButton";
-    sortImage.dataset.status = "default";
-    sortButton.appendChild(sortImage);
-    return sortButton;
+const getCreatedSortButtons = (columnKey) => {
+  const sortButton = createElementWithClassName("button", "js-sortButton");
+  const sortImage = createElementWithClassName("img", "sort-image");
+  sortButton.id = `${columnKey}`;
+  sortImage.dataset.status = "default";
+  sortImage.src = "../img/both.svg";
+  sortImage.alt = "both-image";
+  sortButton.appendChild(sortImage);
+  return sortButton;
 };
 
-const addEventForSortButton = (userContentsData) => {
-    const sortButton = document.getElementById("js-sortButton");
-    sortButton.addEventListener('click', (e) => {
-        const nextStatus = changeSortStatus(e.target.dataset.status);
-        e.target.dataset.status = nextStatus;
-
-        sortContents(userContentsData, nextStatus);        
-    })
-}
-
 const changeSortStatus = (status) => {
-    switch(status){
-        case "default":
-            return "asc";
-        case "asc":
-            return "desc";
-        case "desc":
-        default:
-            return "default";
+  switch (status) {
+    case "default":
+      return "asc";
+    case "asc":
+      return "desc";
+    case "desc":
+    default:
+      return "default";
+  }
+};
+
+const defaultSettingForSortButton = (target, nextStatus) => {
+  const notTarget = [...document.querySelectorAll(".sort-image")].filter(
+    (column) => column !== target
+  );
+  notTarget.forEach((column) => {
+    if (nextStatus !== "default") {
+      column.dataset.status = "default";
+      column.src = "../img/both.svg";
+      column.alt = "both-image";
     }
-}
+  });
+};
+
+const configForSortButton = (target, status) => {
+  switch (status) {
+    case "asc":
+      target.src = "../img/asc.svg";
+      target.alt = "asc-image";
+      break;
+    case "desc":
+      target.src = "../img/desc.svg";
+      target.alt = "desc-image";
+      break;
+    case "default":
```

Comment:
```suggestion
    default:
```

Reply:
Thank you, sir.
[17b2e33](https://github.com/seeeen044/JavaScript_handson/pull/24/commits/17b2e33468c0357284d11586d9d54b3c2320ccca)